### PR TITLE
[TT-8747] [5.1] Plugin docs update

### PR DESCRIPTION
**Pull request by titpetric**
## Related Issue (if applicable)

https://tyktech.atlassian.net/browse/TT-8747

## Description

- documents go workspaces
- adds required `-trimpath` options
- fix one off typo

## Preview Link

https://deploy-preview-2653--tyk-docs.netlify.app/docs/nightly/plugins/supported-languages/golang/#initialise-plugin

## Types of changes

- [ ] Fixing typo (please merge to production)
- [ ] Documenting a new feature (please merge to production)
- [x] Documentation for future release (please do not merge to production)
- [ ] Something else (please add if needs merging to production or not)

## Checklist

- [x] I have [reviewed the guidelines](../CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](../CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I have added a preview link.

### CODE REVIEW

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 1]
```diff
@@ -36,6 +36,34 @@ It's also possible to access the API definition data structure from within a plu
 
 #### Initialise plugin
 
+In Gateway version 5.1, the gateway and plugins transitioned to a go
+module build and don't use vendoring anymore. To create a full workspace
+follow the guide here.
```
Comment (jeffy-mathew) [inline tyk-docs/content/plugins/supported-languages/golang.md]: do we need to link go workspace docs here https://go.dev/ref/mod#workspaces

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 2]
```diff
@@ -36,6 +36,34 @@ It's also possible to access the API definition data structure from within a plu
...
+
+In the parent folder, run the following commands to create a go
+workspace:
+
+```
+go work init ./tyk
+go work use ./tyk_plugin
+```
+
+To build compatible plugins, the gateway dependency that tyk_plugin uses
```
Comment (furkansenharputlu) [inline tyk-docs/content/plugins/supported-languages/golang.md]: `tyk_plugin`

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 3]
```diff
@@ -36,6 +36,34 @@ It's also possible to access the API definition data structure from within a plu
...
+needs to match the checkout in the tyk folder. To get the commit hash, go
+into `tyk` and run `git rev-parse HEAD`. To synchronize the gateway
```
Comment (furkansenharputlu) [inline tyk-docs/content/plugins/supported-languages/golang.md:58]: what if we show the commands in their own line?

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 4]
```diff
@@ -36,6 +36,34 @@ It's also possible to access the API definition data structure from within a plu
 
 #### Initialise plugin
 
+In Gateway version 5.1, the gateway and plugins transitioned to a go
+module build and don't use vendoring anymore. To create a full workspace
+follow the guide here.
+
+You need two checkouts:
+
+1. your plugin sources (using `tyk_plugin`),
```
Comment (dcs3spp) [inline tyk-docs/content/plugins/supported-languages/golang.md]:
```suggestion
1. Your plugin sources (using `tyk_plugin`)
```

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 5]
```diff
@@ -36,6 +36,34 @@ It's also possible to access the API definition data structure from within a plu
...
+2. tyk gateway source for the release (using `tyk`)
```
Comment (dcs3spp) [inline tyk-docs/content/plugins/supported-languages/golang.md]:
```suggestion
2. Tyk gateway source for the release (using `tyk`)
```

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 6]
```diff
@@ -36,6 +36,34 @@ It's also possible to access the API definition data structure from within a plu
 
 #### Initialise plugin
```
Comment (dcs3spp) [inline tyk-docs/content/plugins/supported-languages/golang.md]:
```suggestion
#### Initialise plugin for Gateway 5.1+
```

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 7]
```diff
@@ -36,6 +36,34 @@ It's also possible to access the API definition data structure from within a plu
...
+To build compatible plugins, the gateway dependency that tyk_plugin uses
+needs to match the checkout in the tyk folder. To get the commit hash, go
+into `tyk` and run `git rev-parse HEAD`. To synchronize the gateway
+dependency go into `tyk_plugin` and run `go get
+github.com/TykTechnologies/tyk@<commit-hash>`.
+
+For more information about go workspaces, check out the [official
+tutorial](https://go.dev/doc/tutorial/workspaces).
+
+For versions before 5.1:
```
Comment (dcs3spp) [inline tyk-docs/content/plugins/supported-languages/golang.md]: Create a new section for initialising plugin for gateway < 5.1

```suggestion
#### Initialise plugin for gateway < 5.1
```
 
Currently the tab shows an example sequence of commands for 3.2 etc. Do we need to add tabs for 4.3, 4.2 etc.?

**Review by dcs3spp — CHANGES_REQUESTED** [review]
lgtm
Made minor suggestions, e.g. introduce section headings for initialising plugins 5.1+ and initialising plugin < 5.1

File: tyk-docs/content/plugins/supported-languages/golang.md (hunk 3 above)
Comment (titpetric) [reply tyk-docs/content/plugins/supported-languages/golang.md:58]: I know what you mean, but unconvinced. There's some thing outside of 5.1 docs we're addressing around git commits (e.g. do we link the commits for particular versions and provide docs like the 3.2.1 below, for 4+ 5+?) so the area can be handled there. Let's do the style comment there too :)

File: tyk-docs/content/plugins/supported-languages/golang.md (hunk 1 above)
Comment (titpetric) [reply tyk-docs/content/plugins/supported-languages/golang.md]:
```suggestion
follow the guide below. To find out more about Go workspaces, visit the [official documentation](https://go.dev/ref/mod#workspaces).
```

File: tyk-docs/content/plugins/supported-languages/golang.md (hunk 2 above)
Comment (titpetric) [reply tyk-docs/content/plugins/supported-languages/golang.md]:
```suggestion
To build compatible plugins, the gateway dependency that `tyk_plugin` uses
```

File: tyk-docs/content/plugins/supported-languages/golang.md (hunk 7 above)
Comment (titpetric) [reply tyk-docs/content/plugins/supported-languages/golang.md]: @dcs3spp the commit links issue is known and out of scope, asked for some jira housekeeping to file and link the ticket for that

**Comment by titpetric** [comment]
@dcs3spp @andyo-tyk  - 5.0 is our LTS build, do we need to point it out specifically? What version am I being encouraged to use by default, 5.1 or LTS? Should we maybe rephrase <5.1 as LTS?

**Comment by andyo-tyk** [comment]
I think it's fine to refer to "Tyk version 5.1 or later" and "Prior to Tyk version 5.1"

**Review by dcs3spp — APPROVED** [review]
Looks good to me. Let me know once peers reviewed

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 8]
```diff
@@ -34,7 +34,35 @@ It's also possible to access the API definition data structure from within a plu
...
-#### Initialise plugin
+#### Initialise plugin for Gateway 5.1+
+
+In Gateway version 5.1, the gateway and plugins transitioned to a go
+module build and don't use vendoring anymore. To create a full workspace
+follow the guide below. To find out more about Go workspaces, visit the [official documentation](https://go.dev/ref/mod#workspaces).
+
+You need two checkouts:
+
+1. Your plugin sources (using `tyk_plugin`)
```
Comment (jeffy-mathew) [inline tyk-docs/content/plugins/supported-languages/golang.md:45]: recently we came across an issue when two plugins built with same module name were referenced in different APIs. 
shall we also add a warning to use different module name for different plugins?

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 9]
```diff
@@ -34,7 +34,35 @@ It's also possible to access the API definition data structure from within a plu
...
+go work use ./tyk_plugin
+```
+
+To build compatible plugins, the gateway dependency that `tyk_plugin` uses
+needs to match the checkout in the tyk folder. To get the commit hash, go
+into `tyk` and run `git rev-parse HEAD`. To synchronize the gateway
+dependency go into `tyk_plugin` and run `go get
+github.com/TykTechnologies/tyk@<commit-hash>`.
+
+For more information about go workspaces, check out the [official
```
Comment (jeffy-mathew) [inline tyk-docs/content/plugins/supported-languages/golang.md:62]: duplicate? https://github.com/TykTechnologies/tyk-docs/pull/2653/files#diff-cd04863da6612d95d2230bb7db6708b118fd855623b2e082c8122f01ba881a23R41

Comment (titpetric) [reply tyk-docs/content/plugins/supported-languages/golang.md:62]: It's a different link, one is the docs, another is the tutorial.

File: tyk-docs/content/plugins/supported-languages/golang.md (hunk 8 above)
Comment (titpetric) [reply tyk-docs/content/plugins/supported-languages/golang.md:45]: out of scope, can discuss in docs refinement for plugins area, or as a general errata

**Review by jeffy-mathew — APPROVED** [review]
